### PR TITLE
ENH: Add orthogonalize argument to DCT/DST

### PR DESCRIPTION
**Pull request by peterbell10**
#### Reference issue
Ref gh-15033

#### What does this implement/fix?
This adds an `orthogonalize=None` parameter to the real transforms in `scipy.fft` which controls whether the modified definition of DCT/DST is used without changing the overall scaling. So for example, you can now call
```
dst(..., norm="ortho", orthogonalize=False)
```
for `1/sqrt(N)` scaling of the unmodified DST. Or,
```
dst(..., norm=None, orthogonalize=True)
```
for an unscaled variant of orthogonal DST. Neither of which were possible before.

For backward compatibility, `orthogonalize` defaults to true when `norm="ortho"` and false otherwise.

### CODE REVIEW

**Review by rgommers — APPROVED** [review]
LGTM, thanks @peterbell10. Should be good to merge for `1.8.0`.

@turbulentphil and @mreineck would you like to have a look at this?

**Comment by mreineck** [comment]
Fine with me! The complex interplay between the various flags is a bit unfortunate, but probably unavoidable for compatibility's sake.

I'll try to integrate this change upstream as well, and perhaps also in my other FFT libraries, but it might take some time.

**Review by tylerjereddy — APPROVED** [review]
Ok, this has core developer approval, CI passing with 100 % patch diff coverage, and considerable FFT infrastructure experience between PR author and the reviewers in favor.

Probably merits a small release note entry if you have time to help with that on wiki.

**Comment by tylerjereddy** [comment]
thanks!

**Comment by turbulentphil** [comment]
I just checked the changes, and it seems pretty fine to me. Awesome!
I agree that, due to backwards compatibility, the complexity of flags might be unavoidable.

**Pull request merged**
